Standards/scs-0200-v1-using-sonobuoy-for-kaas-conformance-tests.md

```diff
@@ -0,0 +1,230 @@
+
+## Decision
+
+The KaaS conformance test MUST be provided as a test suite holding the
+test cases for the Kubernetes clusters to be checked.
+Furthermore, the test cases themselves MUST be wrapped by a test framework to:
+
+* handle the creation and deletion of resources
+* collect and present results
+* consolidate redundant code across test cases
+* support the creation of test cases through predefined structures
+
+As with the [k8s-cluster-api-provider][k8s-api] the SCS provides a tooling to generate
+its KaaS infrastructure. Part of the [k8s-cluster-api-provider][k8s-api] is the usage
+of Sonobuoy as a test suite to execute the [Kubernetes own e2e tests][k8s-e2e-tests].
+Investigating those e2e test lead to the conclusion that they are not always reusable.
+The main purpose of Kubernetes own e2e tests is to test the functionality of
+the Kubernetes code itself and not the resources and setup of a specific KaaS infrastructure
+as it is the aim of the SCS KaaS conformance test.
+
+However, considering that the SCS has an ongoing process of defining standards and
+implementing test cases to check their compliance, that means in some cases future standards
+might already be covered by the e2e test inside the Kubernetes repository.
+Hence, before writing tests, a developer SHOULD check the Kubernetes e2e tests for
+existing test cases that might cover conformance of a standard currently being worked on.
+
+As described above, Sonobuoy offers the possibility to generate custom plugins
+that provide self-created test cases. Therefore, future conformance tests MUST
+be executable by Sonobuoy by wrapping them in a Sonobuoy plugin.
+Moreover, three options for the implementation of the SCS KaaS compliance test
+cases are described above so that it can be decided within this decision record which
+options should be used.
+
+As a first decision, "_Option 2_ Go Approach 2: Reuse Kubernetes' own e2e test infrastructure and framework"
+is the least viable, as it would mean copying almost all the files used from
+the Kubernetes e2e tests. This framework is closely linked to the development of
+the Kubernetes code. Therefore, changes to its structure mainly are in line with its usage
+in the Kubernetes repository itself and is most likely not relevant for other parties.
+Changes in the framework cloud have a greater impact on our side as they are
+predictable. The development effort gained by reusing these examples could be
+outweighed by the investment necessary to adapt tests to the corresponding framework changes.
+
+This leaves "_Option 1_ Go approach 1: Pick framework from the Sonobuoy plugin examples"
+and "_Option 3_ Write Python scripts for tests" as methods for implementing test cases.
+It is possible to use both approaches in parallel, which would enable usage of both Python
+and Go code. This could possibly generate two Sonobuoy plugins, but this isn't a necessity.
+
+## Documents
```

Comment:
When the document is rendered, this will appear as a lonely heading, because the references below are not diplayed and instead inserted inline in the places where they are referenced in the text.

```suggestion
```

Reply:
Changed that up @martinmo